[Samarth Verma] iP

vsamarth opened the pull request:
# Duke

**Duke** ensures that you keep track of your todos and deadlines so that you
are not scrambling at the last minute to complete them.

* Open Source and Completely FREE!
* Text based
* Easy to learn
* Completely offline
* **Most Importantly, Extremely Fast** ⚡ 

To use, simply 

1. Ensure that you have [java (at least v11) installed](https://www.oracle.com/java/technologies/downloads/)
2. Download the latest release from [here](https://github.com/vsamarth/ip/releases/download/v0.1/duke.jar).
3. Go to the directory where you downloaded and open a Terminal
4. Run `java -jar duke.jar`
5. Make your life managable!

#### Features

- [x] Managing Tasks
- [x] Saving Tasks
- [x] Searching Tasks
- [ ] View Schedule
- [x] Natural Language Date Parsing


Feel free to examine and help improve the current product by submitting a Pull Request!

vsamarth committed:
Greet user with message

vsamarth committed:
Allow user to add and list tasks

vsamarth committed:
Extract user interface logic

vsamarth committed:
Add support for deadlines and events

vsamarth committed:
Use strings for dates

vsamarth committed:
Allow tasks to be marked completed

vsamarth committed:
Add UI Tests

vsamarth committed:
Allow deleting tasks

vsamarth committed:
Move files into separate packages

vsamarth committed:
Change executable name in test script

vsamarth committed:
Add Gradle support

vsamarth committed:
Allow generating jar files

vsamarth committed:
Add support for jar files and checkstyle

shirsho-12 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,32 @@
+import duke.tasks.TaskList;
+import duke.ui.UserInterface;
+
+public class Duke {
```
Perhaps adding JavaDocs for this class would be helpful.

shirsho-12 commented on src/main/java/duke/commands/Command.java:
```diff
@@ -0,0 +1,9 @@
+import duke.tasks.TaskList;
+import duke.ui.UserInterface;
+
+public abstract class Command {
```
Perhaps adding JavaDocs for this class would be helpful.

shirsho-12 commented on src/main/java/duke/commands/CreateDeadline.java:
```diff
@@ -0,0 +1,23 @@
+import duke.tasks.TaskList;
+import duke.ui.UserInterface;
+
+public class CreateDeadline extends Command {
```
Perhaps adding JavaDocs for this class would be helpful.

shirsho-12 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,102 @@
+        String[] tokens = input.split(" ");
+        String command = tokens[0];
+        String args = input.substring(command.length()).trim();
+        switch (command) {
```
I'd recommend using enums for this.

shirsho-12 commented on src/main/java/duke/ui/UserInterface.java:
```diff
@@ -0,0 +1,62 @@
+        printLine();
+    }
+
+    public void showExitMessage() {
```
Perhaps it'd be a good idea to close the scanner over here. Makes the code more robust.

shirsho-12 reviewed (commented):
Very clean, to-the-point code. Overall, I found your code easy to read and understand. My only gripe with your code is the lack of JavaDocs.

vsamarth committed:
Allow saving tasks to a file

vsamarth committed:
Ignore the data file

vsamarth committed:
Merge branch 'gradle-support'

vsamarth committed:
Add Javadocs for all classes

vsamarth committed:
Fix all style violations

vsamarth committed:
Parse deadlines as dates

vsamarth committed:
Allow finding tasks

vsamarth committed:
Add support for using JavaFx.

Removed all the older text-based testing and ui. Modify the UserInterface class
to act as the AnchorPane.

Adapted from the tutorial provided for JavaFx.

vsamarth committed:
Make old commands work with JavaFx

vsamarth committed:
Add assertions to ensure constraints are met

vsamarth committed:
Merge pull request #4 from vsamarth/branch-A-Assertions

Add assertions to ensure constraints are met

vsamarth committed:
Add missing javadocs

vsamarth committed:
Enable CI using Github Actions

vsamarth committed:
Fix CI build on Windows

vsamarth committed:
Update Readme to reflect features and usage

vsamarth committed:
Add build status badge

vsamarth committed:
Fix CI build script

vsamarth committed:
Parse deadlines using natural language

vsamarth committed:
Parse event times as natural language

vsamarth committed:
Merge pull request #5 from vsamarth/branch-natural-dates

Use natural language for parsing dates

vsamarth committed:
Add features to user guide

vsamarth committed:
Remove unnecessary imports

vsamarth committed:
Merge pull request #6 from vsamarth/branch-A-CodeQuality

Improve Code Quality

vsamarth committed:
Add Unit Tests

vsamarth committed:
Make window unresizable

vsamarth committed:
Fix failing build

vsamarth committed:
Refactor by removing id and using index in TaskList

vsamarth committed:
Add UI Screenshot

vsamarth committed:
Add usage instructions to docs

vsamarth committed:
Add javafx dependency to gradle
